File: trunk-sample/src/main/java/com/capsule/trunk/MainActivity.java
package com.capsule.trunk;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import capsule.trunk.click.ItemClickListener;
import capsule.trunk.decor.HeadDecor;
import com.capsule.trunk.base.BaseActivity;
import com.capsule.trunk.func.anim.AnimActivity;
import com.capsule.trunk.func.click.ClickActivity;
import com.capsule.trunk.func.divider.DividerActivity;
import com.capsule.trunk.func.empty.EmptyActivity;
import com.capsule.trunk.func.expand.ExpandActivity;
import com.capsule.trunk.func.foot.FootActivity;
import com.capsule.trunk.func.head.HeadActivity;
import com.capsule.trunk.func.load.LoadActivity;
import com.capsule.trunk.func.multi.MultipleActivity;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends BaseActivity {

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setTitle("主页");
    initView();
  }

  @Override protected int onGetLayoutId() {
    return R.layout.activity_main;
  }

  private void initView() {
    RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler);
    recyclerView.setLayoutManager(new LinearLayoutManager(this));
    recyclerView.addOnItemTouchListener(new ItemClickListener() {
      @Override public void onItemClick(RecyclerView.ViewHolder vh, int position, View childView) {
        startByPosition(position);
      }
    });
    recyclerView.addItemDecoration(new HeadDecor(R.layout.layout_head, true));

    MainAdapter adapter = new MainAdapter();
    recyclerView.setAdapter(adapter);

    adapter.setData(mockData());
    adapter.notifyDataSetChanged();
  }

  private List<String> mockData() {
    List<String> list = new ArrayList<>();
    list.add("Animator");
    list.add("ClickListener");
    list.add("Divider");
    list.add("EmptyView");
    list.add("Expand");
    list.add("Footer");
    list.add("Header");
    list.add("Level");
    list.add("LoadMore");
    list.add("MultipleItemType");
    list.add("...");
    list.add("...");
    list.add("...");
    list.add("...");
    list.add("...");
    return list;
  }

  private void startByPosition(int position) {
    Intent intent;
    switch (position) {
      case 0:
        intent = new Intent(this, AnimActivity.class);
        break;
      case 1:
        intent = new Intent(this, ClickActivity.class);
        break;
      case 2:
        intent = new Intent(this, DividerActivity.class);
        break;
      case 3:
        intent = new Intent(this, EmptyActivity.class);
        break;
      case 4:
        intent = new Intent(this, ExpandActivity.class);
        break;
      case 5:
        intent = new Intent(this, FootActivity.class);
        break;
      case 6:
        intent = new Intent(this, HeadActivity.class);
        break;
      case 7:
        intent = new Intent(this, LevelActivity.class);
        break;
      case 8:
        intent = new Intent(this, LoadActivity.class);
        break;
      case 9:
        intent = new Intent(this, MultipleActivity.class);
        break;

      default:
        return;
    }
    startActivity(intent);
  }
}

File: frog-sample/src/main/java/com/yuluyao/frog/MainActivity.kt
package com.yuluyao.frog

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.facebook.stetho.Stetho
import com.yuluyao.frog.base.BaseActivity
import com.yuluyao.frog.func.adapter.AdapterActivity
import com.yuluyao.frog.func.databinding.DataBindingActivity
import com.yuluyao.frog.func.divider.DividerActivity
import com.yuluyao.frog.func.touch.ItemTouchActivity
import com.yuluyao.frog.repo.Repo
import kotlinx.android.synthetic.main.activity_main.*
import yuluyao.frog.touch.FrogClickListener
import java.util.*

class MainActivity : BaseActivity() {

  override fun onGetLayoutId(): Int = R.layout.activity_main
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    Repo.init(this )

    title = "主页"
    initView()
    Stetho.initializeWithDefaults(applicationContext)
  }

  private fun initView() {
    recyclerView.layoutManager = LinearLayoutManager(this)
    recyclerView.addOnItemTouchListener(object : FrogClickListener() {
      override fun onItemClicked(position: Int) {
        startByPosition(position)

      }
    })
//    recyclerView.addItemDecoration(HeadDecor(R.layout.layout_head, true))

    val adapter = MenuAdapter()
    recyclerView.adapter = adapter

    adapter.data = mockData()
    adapter.notifyDataSetChanged()
  }

  private fun mockData(): ArrayList<MenuItem> {
    val list = ArrayList<MenuItem>()
    list.add(MenuItem("Adapter",0))
    list.add(MenuItem("DataBinding",0))
    list.add(MenuItem("Item Touch",0))
    list.add(MenuItem("FrogDivider",0))
    list.add(MenuItem("EmptyView",0))
    list.add(MenuItem("Expand",0))
    list.add(MenuItem("Footer",0))
    list.add(MenuItem("Header",0))
    list.add(MenuItem("Level",0))
    list.add(MenuItem("LoadMore",0))
    list.add(MenuItem("MultipleItemType",0))
    list.add(MenuItem("...",0))
    list.add(MenuItem("...",0))
    list.add(MenuItem("...",0))
    list.add(MenuItem("...",0))
    list.add(MenuItem("...",0))
    return list
  }

  private fun startByPosition(position: Int) {
    var intent: Intent? = null
    when (position) {
      0 -> intent = Intent(this, AdapterActivity::class.java)
      1 -> intent = Intent(this, DataBindingActivity::class.java)
      2 -> intent = Intent(this, ItemTouchActivity::class.java)
      3 -> intent = Intent(this, DividerActivity::class.java)
//      3 -> intent = Intent(this, EmptyActivity::class.java)
//      4 -> intent = Intent(this, ExpandActivity::class.java)
//      5 -> intent = Intent(this, FootActivity::class.java)
//      6 -> intent = Intent(this, HeadActivity::class.java)
//      7 -> {
//      }
//      8 -> intent = Intent(this, LoadActivity::class.java)
//      9 -> intent = Intent(this, MultipleActivity::class.java)

      else -> return
    }//intent = new Intent(this, LevelActivity.class);
    startActivity(intent)
  }

  class MenuItem(val title: String, val id: Int)

  class MenuAdapter : RecyclerView.Adapter<MenuAdapter.Holder>() {
    var data: ArrayList<MenuItem> = arrayListOf()
    override fun getItemCount(): Int = data.size
    override fun onCreateViewHolder(parent: ViewGroup, itemType: Int): Holder {
      val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_main, parent, false)
      return Holder(itemView)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
      holder.tvTitle.text = data[position].title
    }

    class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
      val tvTitle: TextView = itemView.findViewById(R.id.tvTitle)
    }
  }
}
